cmd/testgrid-config-generator/main.go

```diff
@@ -182,6 +183,94 @@ type release struct {
 	} `json:"verify"`
 }
 
+// e2ePresubmitJobsByBranch collects statically-defined e2e presubmit jobs (e2e,
+// serial and upgrade) grouped by the targeted branch.
+func e2ePresubmitJobsByBranch(presubmitBranches []string, jobConfig *prowconfig.JobConfig) map[string][]string {
+	presubmitJobs := map[string][]string{}
+	for _, p := range jobConfig.AllStaticPresubmits(nil) {
+		// Only track mandatory jobs
+		if !p.AlwaysRun || p.Optional {
+			continue
+		}
+		// Only track presubmits for repos in the openshift org. This is intended
+		// to minimize the cost of testgrid collection and can be relaxed if
+		// necessary. openshift-priv should not be collected as it rarely sees
+		// presubmit activity.
+		if !strings.HasPrefix(p.Name, "pull-ci-openshift-") ||
+			strings.HasPrefix(p.Name, "pull-ci-openshift-priv") {
+
+			continue
+		}
+		// Group jobs by branch
+		for _, branch := range presubmitBranches {
```

Comment:
The job's configuration has the branches it can be triggered on, let's use that

Reply:
As per the comment below this line, I care about more than the branch. I want to limit skip random repo-local e2e jobs that are typically characterized by not following the naming convention of <branch>-e2e. Really I want to target only jobs that are derived from workflows, but that implementation would be considerably more involved afaict.